programs/gateway/src/lib.rs

```diff
@@ -130,6 +132,45 @@ pub mod gateway {
         Ok(())
     }
 
+    /// Increments nonce, used by TSS in case outbound fails.
+    ///
+    /// # Arguments
+    /// * `ctx` - The instruction context.
+    /// * `amount` - The amount in original outbound.
+    /// * `signature` - The TSS signature.
+    /// * `recovery_id` - The recovery ID for signature verification.
+    /// * `message_hash` - Message hash for signature verification.
+    /// * `nonce` - The current nonce value.
+    pub fn increment_nonce(
```

Comment:
Can't we use a name describing the purpose on the high-level? like `report_fail_outbound`

Incrementing nonce is more the implementation detail, is there another conidtion where you would just increase the nonce?

Reply:
this name looks fine to me
if want something more specific to our use of it, maybe "cancel_outbound" . 

A more drastic thing to do is one can consider simply reusing/abusing the withdraw to increment the nonce-- just set amount to 0 lamport if that works.  Then we don't even need this PR 😄